src/System.Text.Json/src/System/Text/Json/JsonReader.cs

```diff
@@ -59,7 +62,7 @@ public void Dispose()
 
         public bool Read()
         {
-            var canRead = _index < _str.Length;
```

Comment:
Why this change? Is Utf8String.Length expensive? If yes, can we fix it in Utf8String?